### PR TITLE
perf: prepare to skip zkey parsing

### PR DESCRIPTION
**Pull request by chokobole**
# Description

Rapidsnark uses a pointer instead of real parsing. Since the AffinePoint of Rapidsnark only contains x and y, we can't adopt this approach. Unlike Rapidsnark and Halo2, which don't include an infinity value, our implementation derived from Arkworks does have an infinity value. This necessitates parsing the ZKey. Additionally, when (de)serializing a zero AffinePoint, it's easy to forget to consider the infinity flag, so this PR mainly removes this flag.

However, this PR is preparatory because there is an issue when performing operations on the AffinePoint from the memory read from the file, causing a segmentation fault. This issue must be addressed to create proofs as quickly as Rapidsnark.

### CODE REVIEW

File: benchmark/msm/arkworks/Cargo.toml [hunk 1]
```diff
@@ -18,6 +18,7 @@ publish = false
 
 [dependencies]
 ark-bn254 = "0.4.0"
+ark-ff = { version = "^0.4.0", features = ["asm"] }
```
Comment (TomTaehoonKim) [inline benchmark/msm/arkworks/Cargo.toml:21]: Can you make this a separate commit?

**Review by Insun35 — APPROVED** [review]
I think the commit message of 32f8c31 should be chore.
Else, LGTM

File: tachyon/c/math/elliptic_curves/generator/point.h.tpl [hunk 2]
```diff
@@ -20,12 +20,9 @@
  *
```
Comment (ashjeong) [inline tachyon/c/math/elliptic_curves/generator/point.h.tpl:20]: Please change the description to remove mentions of the boolean flag.

**Review by ashjeong — CHANGES_REQUESTED** [review]

File: benchmark/msm/arkworks/Cargo.toml (hunk 1 above)
Comment (chokobole) [reply benchmark/msm/arkworks/Cargo.toml:21]: Actually this change is a part of this commit! Since `infinity` is removed and this commit needs a trait from `ark-ff` to call `is_zero` method.

**Comment by chokobole** [comment]
> I think the commit message of [32f8c31](https://github.com/kroma-network/tachyon/commit/32f8c31e101fdfd88955edd56daebe3dc05867f9) should be chore. Else, LGTM

If I read the type definition from https://github.com/kroma-network/.github/blob/main/CONTRIBUTING.md

> style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc). Other examples include sorting build targets or headers.

`style` is more proper type! Thanks for pointing out it

File: benchmark/msm/arkworks/src/lib.rs [hunk 3]
```diff
@@ -19,10 +20,12 @@ pub extern "C" fn run_msm_arkworks(
...
         bases_vec.reserve_exact(bases.len());
         for i in 0..size {
+            let x = mem::transmute(bases[i].x);
+            let y = mem::transmute(bases[i].y);
             bases_vec.push(G1Affine {
-                x: mem::transmute(bases[i].x),
-                y: mem::transmute(bases[i].y),
-                infinity: bases[i].infinity,
+                x,
+                y,
```
[the remark this answers is not in the document]
Comment (chokobole) [reply benchmark/msm/arkworks/src/lib.rs:27]: I needed to call `is_zero()` method to check whether the point is point at infinity. Since it is defined by arkworks, so I needed to `transmute` to call this method.

File: tachyon/c/math/elliptic_curves/msm/msm_input_provider.h [hunk 4]
```diff
@@ -33,24 +32,21 @@ class MSMInputProvider {
...
       }
+      bases_ = bases_owned_;
     } else {
-      bases_owned_.resize(size);
-    }
-    OPENMP_PARALLEL_FOR(size_t i = 0; i < size; ++i) {
-      bases_owned_[i] =
-          AffinePoint(points[i], points[i].x.IsZero() && points[i].y.IsZero());
+      bases_ = absl::MakeConstSpan(
+          reinterpret_cast<const AffinePoint*>(bases_in), size);
```
[the remark this answers is not in the document]
Comment (chokobole) [reply tachyon/c/math/elliptic_curves/msm/msm_input_provider.h:48]: The `CPoint` structure has only `x` and `y` members. Previously, `AffinePoint` had additional `infinity` members, the size of structures were different. But now the structure of those class became same, so it's possible to make `bases_` just point to `CPoint` array!

[the remark this answers is not in the document]
Comment (ashjeong) [reply tachyon/c/math/elliptic_curves/msm/msm_input_provider.h:48]: Through internal discussion, I've learned that `bases_owned_` is a member variable used to provide lasting memory for `absl::span bases_`, and the values inside `bases_owned_` are unneeded if `absl::span bases_` does not rely on `bases_owned_`!

**Review by ashjeong — APPROVED** [review]
LGTM

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by dongchangYoo — APPROVED** [review]
LGTM
